bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BallerinaXMLSerializer.java

```diff
@@ -0,0 +1,324 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.jvm;
+
+import org.ballerinalang.jvm.util.exceptions.BallerinaException;
+import org.ballerinalang.jvm.values.ErrorValue;
+import org.ballerinalang.jvm.values.XMLComment;
+import org.ballerinalang.jvm.values.XMLItem;
+import org.ballerinalang.jvm.values.XMLPi;
+import org.ballerinalang.jvm.values.XMLSequence;
+import org.ballerinalang.jvm.values.XMLText;
+import org.ballerinalang.jvm.values.api.BXML;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+
+/**
+ * XML Serializer for Ballerina XML value trees.
+ *
+ * @since 1.1.0
+ */
+public class BallerinaXMLSerializer extends OutputStream {
+    private static final XMLOutputFactory xmlOutputFactory = XMLOutputFactory.newInstance();
+    private XMLStreamWriter xmlStreamWriter;
+    private Deque<Set<String>> parentNSSet;
+    private int nsNumber;
+
+
+    public BallerinaXMLSerializer(OutputStream outputStream) {
+        try {
+            xmlStreamWriter = xmlOutputFactory.createXMLStreamWriter(outputStream);
+            parentNSSet = new ArrayDeque<>();
+        } catch (XMLStreamException e) {
+            throw new BallerinaException(e);
+        }
+    }
+
+    @Override
+    public void write(int b) {
+        assert false;
+    }
+
+    @Override
+    public void flush() {
+        try {
+            xmlStreamWriter.flush();
+        } catch (XMLStreamException e) {
+            throw new BallerinaException(e);
+        }
+    }
+
+    @Override
+    public void close() throws IOException {
+        try {
+            xmlStreamWriter.close();
+        } catch (XMLStreamException e) {
+            throw new BallerinaException(e);
+        }
+    }
+
+    public void write(BXML xmlValue) {
```

Comment:
What if we define `throws` since this a public method?

Reply:
It doesn't throw any checked exceptions, in that case, do we need to do that.